Userland/Libraries/LibWeb/HTML/HTMLProgressElement.cpp

```diff
@@ -17,4 +19,57 @@ HTMLProgressElement::~HTMLProgressElement()
 {
 }
 
+RefPtr<Layout::Node> HTMLProgressElement::create_layout_node(NonnullRefPtr<CSS::StyleProperties> style)
+{
+    return adopt_ref(*new Layout::Progress(document(), *this, move(style)));
+}
+
+double HTMLProgressElement::value() const
+{
+    auto parsedValue = strtod(attribute(HTML::AttributeNames::value).characters(), nullptr);
+    if (!isfinite(parsedValue) || parsedValue < 0)
+        return 0;
+
+    return min(parsedValue, max());
+}
+
+void HTMLProgressElement::set_value(double value)
+{
+    if (value < 0)
+        return;
+
+    set_attribute(HTML::AttributeNames::value, String::formatted("{}", value));
+
+    if (layout_node())
+        layout_node()->set_needs_display();
+}
+
+double HTMLProgressElement::max() const
+{
+    auto parsedValue = strtod(attribute(HTML::AttributeNames::max).characters(), nullptr);
+    if (!isfinite(parsedValue) || parsedValue <= 0)
+        return 1;
+
+    return parsedValue;
+}
+
+void HTMLProgressElement::set_max(double value)
+{
+    if (value <= 0)
+        return;
+
+    set_attribute(HTML::AttributeNames::max, String::formatted("{}", value));
```

Comment:
This could use `String::number` (even though it does the exact same thing as this does)